dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramInstanceService.java

```diff
@@ -112,23 +112,7 @@ public long addProgramInstance( ProgramInstance programInstance )
     @Transactional
     public void deleteProgramInstance( ProgramInstance programInstance )
     {
-        deleteProgramInstance( programInstance, false );
-    }
-
-    @Override
-    @Transactional
-    public void deleteProgramInstance( ProgramInstance programInstance, boolean forceDelete )
-    {
-        if ( forceDelete )
-        {
-            programInstanceStore.delete( programInstance );
-        }
-        else
-        {
-            programInstance.setDeleted( true );
-            programInstance.setStatus( ProgramStatus.CANCELLED );
-            programInstanceStore.update( programInstance );
-        }
+        programInstanceStore.delete( programInstance );

```

Comment:
Are you sure there are no side effects of removing setting the status to CANCELLED ? Maybe you can set it before you send it onwards to the deletion store?